File: python_stack/python/OOP/Bank Account/Bank_Account.py
class BankAccount:
    def __init__(self, int_rate = 0.01, balance = 0):
        self.int_rate = int_rate
        self.balance = balance

    def deposit(self, amount):
        self.balance += amount
        return self

    def withdraw(self, amount):
        self.balance -= amount
        return self

    def display_account_info(self):
        print("Balance: $"+str(self.balance))

    def yield_interest(self):
        if self.balance>0:
            self.balance += self.balance * self.int_rate
        return self

account1 = BankAccount(0.02,500)
account2 = BankAccount()

account1.deposit(300).deposit(100).deposit(50).withdraw(730).yield_interest().display_account_info()

account2.deposit(300).deposit(100).withdraw(50).withdraw(100).withdraw(70).withdraw(300).yield_interest().display_account_info()

File: Java/OOP/BankAccount/BankAccount.java
public class BankAccount {
    private double checkingBalance;
    private double savingsBalance;
    static int numOfAccounts = 0;
    static double totalMoney = 0.0;

    // Constructor:
    public BankAccount(){
        numOfAccounts++;
    }

    //getters:
    public double getCheckingBalance(){
        return checkingBalance;
    }
    public double getSavingsBalance(){
        return savingsBalance;
    }

    public void deposit(Character a, double amount){
        if (a.equals('c')){
            checkingBalance += amount;
        }
        else if (a.equals('s')){
            savingsBalance += amount;
        }
        totalMoney += amount;
    }

    public void withdraw(Character a, double amount){
        if (a.equals('c')){
            if (amount > checkingBalance){
                System.out.println("insufficient funds");
            }
            else{
                checkingBalance -= amount;
            }
        }
        else if (a.equals('s')){
            if (amount > savingsBalance){
                System.out.println("insufficient funds");
            }
            else{
                savingsBalance -= amount;
            }
        }
    }

    public void displayTotal(){
        System.out.println("Checking Balance - $" + checkingBalance);
        System.out.println("Savings Balance - $" + savingsBalance);
        System.out.println("Total of both balances - $" + (checkingBalance + savingsBalance));
    }
}
